crates/fluvio-sc-schema/src/smartmodule/mod.rs

```diff
@@ -34,4 +34,21 @@ mod convert {
 
     ObjectTryFrom!(WatchResponse, SmartModule);
     ObjectTryFrom!(ListResponse, SmartModule);
+
+    use super::SmartModuleMetadataSpec;
+
+    impl AdminSpec for SmartModuleMetadataSpec {
```

Comment:
No need to implement AdminSpec.  It should be associated type of SmartModuleSpec

Reply:
I think it's needed to used for [`ListResponse`](https://github.com/infinyon/fluvio/blob/master/crates/fluvio-sc-schema/src/objects/list.rs#L19-L26)

Reply:
that's ListRequest

Reply:
I mean it's required for both though `ListRequest` and `ListResponse`.

Reply:
```
 pub trait AdminSpec: Spec + Encoder + Decoder {
        type ListFilter: Encoder + Decoder + Sized + Debug;
        type ListType: Encoder + Decoder + Debug;
        type WatchResponseType: Spec + Encoder + Decoder;
    }
```

The only requirement for ListType is that it can be serialized/deserialized thru Fluvio Encoder